### PR TITLE
User contact form

### PR DESCRIPTION
**Pull request by alexandraciobica**
Pull Request for Issue https://github.com/joomla-projects/gsoc18_enhance_users/issues/8.

### Summary of Changes
Added the contact form to the user, with both global and local configuration of the form settings.

### CODE REVIEW

File: administrator/components/com_users/tmpl/users/default_batch_footer.php [hunk 1]
```diff
@@ -9,8 +9,6 @@
 defined('_JEXEC') or die;
 use Joomla\CMS\Language\Text;
 
-use Joomla\CMS\Language\Text;
```
Comment (laoneo) [inline administrator/components/com_users/tmpl/users/default_batch_footer.php]: Should this one be done against the main repo? If not, then better to remove line 10 as a new line should be after the jexec test.

File: administrator/components/com_users/config.xml [hunk 2]
```diff
@@ -184,6 +184,113 @@
 		/>
 	</fieldset>
 
+	<fieldset
+			name="Contact_Form"
```
Comment (laoneo) [inline administrator/components/com_users/config.xml]: I guess there are one tab too much on every new line.

File: components/com_users/Controller/UserController.php [hunk 3]
```diff
@@ -8,24 +8,42 @@
  */
 namespace Joomla\Component\Users\Site\Controller;
 
-defined('_JEXEC') or die;
```
Comment (laoneo) [inline components/com_users/Controller/UserController.php]: Jexec test needs to be done as first code after the namespace declaration.

File: components/com_users/Model/UserModel.php [hunk 4]
```diff
@@ -10,18 +10,29 @@
...
+class UserModel extends FormModel
 {
+	/**
+	 * A loaded item
+	 *
+	 * @since   1.6
+	 */
+	protected $_item = null;
+
+	protected $user;
```
Comment (laoneo) [inline components/com_users/Model/UserModel.php]: Missing doc bloc.

File: components/com_users/layouts/field/render.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
+<?php
```
Comment (laoneo) [inline components/com_users/layouts/field/render.php:1]: Do we really need this file, I'm not sure.

File: components/com_users/layouts/fields/render.php [hunk 6]
```diff
@@ -0,0 +1,77 @@
+<?php
```
Comment (laoneo) [inline components/com_users/layouts/fields/render.php:1]: Do we really need this file, I'm not sure.

File: components/com_users/tmpl/user/default.php [hunk 7]
```diff
@@ -8,6 +8,7 @@
  */
 
 defined('_JEXEC') or die;
+$params = $this->item->params;
```
Comment (laoneo) [inline components/com_users/tmpl/user/default.php]: New line before.

File: components/com_users/tmpl/user/default.php [hunk 8]
```diff
@@ -23,6 +24,10 @@
 	<div itemprop="email"><?php echo $this->escape($this->item->username); ?></div>
 	<div><?php echo $this->escape($this->item->email); ?></div>
 
+	<?php  if ($params->get('show_email_form')) : ?>
```
Comment (laoneo) [inline components/com_users/tmpl/user/default.php]: Only one space before if.

File: administrator/components/com_users/tmpl/users/default_batch_footer.php (hunk 1 above)
Comment (alexandraciobica) [reply administrator/components/com_users/tmpl/users/default_batch_footer.php]: It shouldn't be done against main repo, this was due to a merge that didn't conflict but didn't merge ok either. I added the newline.

File: components/com_users/layouts/field/render.php (hunk 5 above)
Comment (alexandraciobica) [reply components/com_users/layouts/field/render.php:1]: I  am not sure either, but the mail looks like this without the render
screen shot 2018-07-17 at 12 17 17

**Pull request merged**